applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element.cpp

```diff
@@ -231,14 +242,396 @@ void EmbeddedFluidElement<TBaseElement>::NormalizeInterfaceNormals(
 }
 
 template <class TBaseElement>
-void EmbeddedFluidElement<TBaseElement>::AddBoundaryConditionPenaltyContribution(
+void EmbeddedFluidElement<TBaseElement>::AddSlipNormalPenaltyContribution(
+    MatrixType& rLHS,
+    VectorType& rRHS,
+    const EmbeddedElementData& rData) const {
+
+    // Obtain the previous iteration velocity solution
+    array_1d<double,LocalSize> values;
+    this->GetCurrentValuesVector(rData,values);
+
+    // If there is embedded velocity, substract it to the previous iteration solution
+    if (this->Has(EMBEDDED_VELOCITY)) {
+        const array_1d<double, 3 >& embedded_vel = this->GetValue(EMBEDDED_VELOCITY);
+        array_1d<double, LocalSize> embedded_vel_exp = ZeroVector(LocalSize);
```

Comment:
`array_1d<double, LocalSize> embedded_vel_exp(LocalSize,0.0)` has the same effect and avoids the `ZeroVector()` call. (No idea of how much more expensive it is, though)

Reply:
Done!